app/models/solidus_stripe/gateway.rb

```diff
@@ -29,38 +29,6 @@ def initialize(options)
 
     attr_reader :client
 
-    # Authorizes a certain amount on the provided payment source.
-    #
-    # We create and confirm the Stripe payment intent in two steps. That's to
-    # guarantee that we associate a Solidus payment on the creation, and we can
-    # fetch it after the webhook event is published on confirmation.
-    #
-    # The Stripe payment intent id is copied over the Solidus payment
-    # `response_code` field.
-    def authorize(amount_in_cents, source, options = {})
```

Comment:
I am concerned about removing the gateway methods. Are we sure there's no situation where someone could call `#authorize` from the backend? Also, they might be helpful as library code. Otherwise, we're coupling the implementation to SSF to a great extent.

Reply:
> Otherwise, we're coupling the implementation to SSF to a great extent.

I think this has more to do with how stripe works, which has the authorization done as part of the in-browser confirmation.

Any other use of the Stripe API should be accessed through the `stripe` gem, possibly using the credentials provided by the payment method with `payment_method.gateway.request {…}`.

@waiting-for-dev you see any other case in which having `#authorize` available in the gateway would be useful?

Reply:
Yes, there's not a lot we can do here. We could keep these methods for those who want to use other flows (maybe some more elaborate wallet implementation with subscriptions?) but it's a bit speculating at this point.

Reply:
Ok, let's remove them, and we can always add them later if needed.

Reply:
That said, a noop with a comment (and maybe a simple implementation example) would not be bad!

Reply:
Yes, the simple implementation example I meant was as a comment or documentation, being clear it's some form of pseudo-code. But maybe it's just a matter of linking the stripe documentation?

Reply:
@kennyadsl it's possible that authorize/purchase methods will be required by https://github.com/solidusio/solidus_stripe/issues/266.

Can we defer this discussion to that ticket? Is that fine?

Reply:
Sure, but can we at least open an issue so that if that's not covered by #266, we don't forget?

Reply:
Absolutely, here it is: https://github.com/solidusio/solidus_stripe/issues/267